test/gtest/uct/ib/test_ib.cc

```diff
@@ -18,7 +17,7 @@ test_uct_ib::test_uct_ib() : uct_test() {
     memset(&m_port_attr, 0, sizeof(m_port_attr));
 }
 
-void test_uct_ib::init() {
+void test_uct_ib::ib_init() {
```

Comment:
still, why do we need to separate ib_init() from init()?
maybe need another test class, like test_ib_with_specific_port, which also tries to find and open GetParam()->dev_name?

Reply:
> still, why do we need to separate ib_init() from init()?

only `test_uct_ib_lmc` and `test_uct_ib_gid_idx` test cases require direct device initialization to perfrom checks
`test_uct_ib::ib_init()` was added to `test_uct_ib` class to avoid duplication in `test_uct_ib`  (`test_uct_ib_lmc` and `test_uct_ib_gid_idx` successors, other successors don't nned this)

> maybe need another test class, like test_ib_with_specific_port, which also tries to find and open GetParam()->dev_name?

do you mean to create the following hierarchical structure:
```
                            test_uct_ib
                          /            \
test_uct_ib_with_specific_port        ... (they don't open device directly)
(does actions from `ib_init`
 inside `init`)
   /                \
test_uct_ib_lmc     test_uct_ib_gid_idx
```

Reply:
yes, like you wrote above

Reply:
> yes, like you wrote above

@yosefe could you review new approach, pls?

`test_uct_ib_with_specific_port` is an interface, has two pure virtual methods used in `test_uct_ib_with_specific_port::init()`:
- `bool should_skip()`
- `void skip()`

`test_uct_ib_lmc`, `test_uct_ib_gid_idx` and `test_uct_ib_pkey` implements `should_skip` and `skip`.
Also, `test_uct_ib_pkey` re-implements `test_uct_ib::send_recv_short()` in the following way (it was in previous patch - 1st commit):
```
void test_uct_ib_with_specific_port::send_recv_short() {
    test_uct_ib::init();
    test_uct_ib::send_recv_short();
    test_uct_ib::cleanup()
}
```
This is needed to accept new pkey values when we do testing in a loop over all available pkeys

But now, we need to call `test_uct_ib::cleanup()` after `test_uct_ib_with_specific_port::init()` in `test_uct_ib_pkey::init` to avoid overwriting UCT entries, and then call `test_uct_ib:init()` before `test_uct_ib_with_specific_port::cleanup()` in `test_uct_ib_pkey::cleanup`